ElevatorSim/src/elevatorsim/floor/Floor.java

```diff
@@ -69,4 +69,22 @@ public void loadPassengers(Direction directionLamp) {
 	public Integer getFloorNumber() {
 		return floorNumber;
 	}
+	
+	/**
+	 * Get the list of going up messages from this floor
+	 * @return the list of messages
+	 */
+	public List<MessageRequest> getActiveUpRequests() {
+		List<MessageRequest> list = activeUpRequests;
+		return list;
```

Comment:
I'd prefer this to be an ImmutableList rather than the actual array list from the floor.

Try Collections.unmodifiableList(activeUpRequests);

Reply:
fixed